[DatePicker] inline property for none dialog picker

tyfoo opened the pull request:
Inline DatePicker, supports the same `API` as the default DatePicker (dialog)

```<DatePicker inline={true} autoOk={true} />```

Implements #1822

otroboe commented:
:+1:

oliviertassinari commented:
@tyfoo I think that an inlined DatePicker as TimerPicker would be a good feature. What do you think about using https://github.com/callemall/material-ui/pull/2043 to do so?

tyfoo commented:
@oliviertassinari Sure. I don't mind using the `Popover`, but does that mean I need to wait for #2043 to be merged?

May I suggest using both options, i.e. `inline` and `popover`? I'll be happy to make any changes needed.

oliviertassinari commented:
> May I suggest using both options, i.e. inline and popover? I'll be happy to make any changes needed.

Sounds like a good idea.

oliviertassinari commented on src/date-picker/date-picker-dialog.jsx:
```diff
@@ -44,6 +45,7 @@ const DatePickerDialog = React.createClass({
     wordings: React.PropTypes.object,
     disableYearSelection: React.PropTypes.bool,
     initialDate: React.PropTypes.object,
+    inline: React.PropTypes.bool,
```
Should we add `inline: false` to the `getDefaultProps`?

oliviertassinari replied:
Can you add the defaut value to `getDefaultProps`?

oliviertassinari replied:
I would rather not use a bool type, since we may add the popover container. Maybe we should call this property `container` with a value between `dialog` (default) and `inline`. Soon `popover`.

oliviertassinari commented on src/date-picker/date-picker-inline.jsx:
```diff
@@ -0,0 +1,50 @@
+    let {
+      style,
+      actions,
+      display,
```
can we call it `open`?

oliviertassinari commented on src/date-picker/date-picker-dialog.jsx:
```diff
@@ -148,18 +152,21 @@ const DatePickerDialog = React.createClass({
       );
     }
-
+    let dialogProps = Object.assign({}, other, {
```
Why not passing the properties using the JSX syntax?

tyfoo commented:
@oliviertassinari Thanks for all your feedback and suggestions. I've made the changes needed (to prepare for `Popover`) and amended my last commit.

Please let me know if there is anything else I can do.

oliviertassinari commented on src/date-picker/date-picker-inline.jsx:
```diff
@@ -0,0 +1,56 @@
+        zIndex: 100,
+        width: '100%',
+        position: 'relative',
+        display: (open === true ? 'block' : 'none'),
```
Instead of using the display property, why not doing in the render method?
This would prevent to render the Calander when it's hidden. Though?

tyfoo replied:
I like that idea. I'll make the needed changes.

oliviertassinari commented on src/date-picker/date-picker-inline.jsx:
```diff
@@ -0,0 +1,56 @@
+      ...other,
+    } = this.props;
+
+    let styles = {
```
You can use `const` here.

oliviertassinari commented on src/date-picker/date-picker-inline.jsx:
```diff
@@ -0,0 +1,56 @@
+    } = this.props;
+
+    let styles = {
+      actions: {
```
In the inline mode, there is a missing margin at the bottom of the action.

oliviertassinari commented on src/date-picker/date-picker-inline.jsx:
```diff
@@ -0,0 +1,56 @@
+      style,
+      actions,
+      open,
+      ...other,
```
`children` is missing here.

oliviertassinari commented:
@tyfoo We are almost there :smile: .

tyfoo commented:
@oliviertassinari Thanks again for your feedback and suggestions. I've made the changes and amended my last commit. Let me know if you have any other thoughts or changes you'd like me to do.

oliviertassinari commented:
@tyfoo We have updated the DatePicker to use the new Dialog API. This add some conflict with your PRs. Sorry.

tyfoo commented:
@oliviertassinari Thank you. I have merged + squashed from master. This PR should be ready to merge, but let me know if there is anything I need to do.

oliviertassinari commented:
@tyfoo The diff is weird: `+1,819 −542`

tyfoo commented:
@oliviertassinari I rebased from master and squashed the commits. I can undo my last commit (the merge), but what should I do differently this time? Any suggestions?

oliviertassinari commented:
@tyfoo Thanks for this, I'm on it :+1:
